make-rules/shared-macros.mk

```diff
@@ -540,22 +621,61 @@ PERL_ARCH_FUNC=	$(shell $(1) -e 'use Config; print $$Config{archname}')
 PKG_MACROS +=   PERL_ARCH=$(PERL_ARCH)
 PKG_MACROS +=   PERL_VERSION=$(PERL_VERSION)
 
+# Config magic for Postgres/EnterpriseDB/...
+# Default DB version is the oldest one, for hopefully best built complatibility
 PG_VERSION ?=   9.3
+PG_IMPLEM ?=    postgres
 PG_VERNUM =     $(subst .,,$(PG_VERSION))
-PG_HOME =       /usr/postgres/$(PG_VERSION)
+# For dependencies, including REQUIRED_PACKAGES if needed
+PG_BASEPKG =    database/$(PG_IMPLEM)-$(PG_VERNUM)
+
+PG_HOME =       $(USRDIR)/$(PG_IMPLEM)/$(PG_VERSION)
 PG_BINDIR.32 =  $(PG_HOME)/bin
 PG_BINDIR.64 =  $(PG_HOME)/bin/$(MACH64)
+PG_BINDIR =     $(PG_BINDIR.$(BITS))
 PG_INCDIR =     $(PG_HOME)/include
 PG_MANDIR =     $(PG_HOME)/man
 PG_SHAREDIR =   $(PG_HOME)/share
 PG_DOCDIR =     $(PG_HOME)/doc
 PG_LIBDIR.32 =  $(PG_HOME)/lib
 PG_LIBDIR.64 =  $(PG_HOME)/lib/$(MACH64)
+PG_LIBDIR =     $(PG_LIBDIR.$(BITS))
 PG_CONFIG.32 =  $(PG_BINDIR.32)/pg_config
 PG_CONFIG.64 =  $(PG_BINDIR.64)/pg_config
+PG_CONFIG =     $(PG_CONFIG.$(BITS))
 
 PKG_MACROS +=   PG_VERSION=$(PG_VERSION)
 PKG_MACROS +=   PG_VERNUM=$(PG_VERNUM)
+PKG_MACROS +=   PG_BASEPKG=$(PG_BASEPKG)
+
+# Config magic for MySQL/MariaDB/Percona/...
+# Default DB version is the oldest one, for hopefully best built compatibility
+# NOTE: At this time the gate does not provide a recipe for actual "mysql"
+# The "/usr/mysql/*" trees are mediated to preferred MariaDB or Percona variant
+MY_VERSION ?=   5.5
```

Comment:
FWIW, I too am not a fan of "MY_" in a generic position like this - just
copied "PG_" lines conveniently. "MYSQL_" might be an option if the name is
acceptable to cover mariadb and percona too. And if the difference in
varname lengths is not bugging anyone else ;p

Or perhaps add DB to both (but then rewrite all PG_ recipes to match)?
DB_PG_* or MY_DB_* ....
--
Typos courtesy of GMail on my Samsung Android

18.07.2016 7:23 пользователь "Alexander Pyhalov" <notifications@github.com>
написал:

> In make-rules/shared-macros.mk
> <https://github.com/OpenIndiana/oi-userland/pull/2027#discussion_r71101231>
> :
>
> >
> >  PKG_MACROS +=   PG_VERSION=$(PG_VERSION)
> >  PKG_MACROS +=   PG_VERNUM=$(PG_VERNUM)
> > +PKG_MACROS +=   PG_BASEPKG=$(PG_BASEPKG)
> > +
> > +# Config magic for MySQL/MariaDB/Percona/...
> > +# Default DB version is the oldest one, for hopefully best built compatibility
> > +# NOTE: At this time the gate does not provide a recipe for actual "mysql"
> > +# The "/usr/mysql/*" trees are mediated to preferred MariaDB or Percona variant
> > +MY_VERSION ?=   5.5
>
> I like the idea, but don't like prefix, too generic, high chances for
> intersection with per-component flags (we already have MY_(LD|C)FLAGS,
> MY_PATH, MY_LDFLAGS... @xen0l <https://github.com/xen0l> , what do you
> think?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/2027/files/84ddcf536f674b3f0b0f9ad47908b86100bc01a3#r71101231>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABj5lOywsOkcb_VfJ4BPjEfOserBG07tks5qWw26gaJpZM4Ij_l9>
> .
>